libraries/chain/backing_store/db_context_rocksdb.cpp

```diff
@@ -218,11 +218,11 @@ namespace eosio { namespace chain { namespace backing_store {
          payer = old_payer;
       }
 
-      const payer_payload old_pp{*old_key_value.value};
-      const auto old_value_actual_size = old_pp.value_size;
+      const auto old_value_actual_size = key_store.value->size();
 
       const payer_payload pp{payer, value, value_size};
-      set_value(old_key_value.full_key, pp);
+      prefix_bundle old_key = get_primary_slice_in_primaries(table_store.contract, table_store.scope,table_store.table, key_store.primary);
```

Comment:
get_primary_slice_in_primaries creates an extra key that we don't use, so we could create a new method that just creates the key_bundle instead of a prefix_bundle.

Reply:
Thanks for your suggestion. Will create a new method in a future PR in the performance activity.